### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: tests/hikari/impl/test_entity_factory.py [hunk 1]
```diff
@@ -2584,7 +2590,7 @@ def message_payload(
         self, user_payload, member_payload, custom_emoji_payload, partial_application_payload, embed_payload
     ):
         del member_payload["user"]
-        return {
+        payload = {
```
Comment (FasterSpeeding) [inline tests/hikari/impl/test_entity_factory.py:2593]: What's the reasoning behind this change? 👀

File: tests/hikari/impl/test_entity_factory.py [hunk 2]
```diff
@@ -2912,18 +2944,21 @@ def test_deserialize_message_with_null_and_unset_fields(
         assert message.activity is None
         assert message.application is None
         assert message.message_reference is None
+        assert message.referenced_message is undefined.UNDEFINED
         assert message.stickers == []
         assert message.nonce is None
 
-    def test_deserialize_message_with_unset_application_fields(self, entity_factory_impl, message_payload):
+    def test_deserialize_message_with_unset_fields(self, entity_factory_impl, message_payload):
```
Comment (FasterSpeeding) [inline tests/hikari/impl/test_entity_factory.py:2951]: with_other_unset_fields?

File: hikari/impl/entity_factory.py [hunk 3]
```diff
@@ -986,11 +986,13 @@ def deserialize_welcome_screen(self, payload: data_binding.JSONObject) -> guild_
...
             emoji: typing.Optional[emoji_models.Emoji] = None
-            if emoji_id is not None and emoji_name is not None:
-                emoji = emoji_models.CustomEmoji(id=snowflakes.Snowflake(emoji_id), name=emoji_name, is_animated=None)
-
-            elif emoji_name is not None:
-                emoji = emoji_models.UnicodeEmoji(emoji_name)
+            if emoji_name is not None:
+                if emoji_id is not None:
+                    emoji = emoji_models.CustomEmoji(
+                        id=snowflakes.Snowflake(emoji_id), name=emoji_name, is_animated=None
```
[the remark this answers is not in the document]
Comment (davfsa) [reply hikari/impl/entity_factory.py:992]: Idea was to remove checking `emoji_name is not None` twice

File: tests/hikari/impl/test_entity_factory.py (hunk 1 above)
Comment (davfsa) [reply tests/hikari/impl/test_entity_factory.py:2593]: Forgot to revert this. My initial fix to add `referenced_message` was to have it reference to the same payload, lol

File: hikari/impl/entity_factory.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (FasterSpeeding) [reply hikari/impl/entity_factory.py:992]: If you want the slightly branchier approach then go ahead ig but `if x is None` checks are fast enough that it shouldn't really matter if you make one twice

File: tests/hikari/impl/test_event_manager.py [hunk 4]
```diff
@@ -29,9 +32,34 @@
 from hikari import intents
 from hikari import presences
 from hikari.impl import event_manager
+from hikari.internal import time
 from tests.hikari import hikari_test_helpers
 
 
+def test_fixed_size_none():
```
Comment (davfsa) [inline tests/hikari/impl/test_event_manager.py:39]:
```suggestion
def test_fixed_size_nonce():
```

**Review by FasterSpeeding — APPROVED** [review]

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**